qjs.c

```diff
@@ -437,10 +437,10 @@ int main(int argc, char **argv)
         fprintf(stderr, "qjs: cannot allocate JS runtime\n");
         exit(2);
     }
-    if (memory_limit != 0)
-        JS_SetMemoryLimit(rt, memory_limit);
-    if (stack_size != 0)
-        JS_SetMaxStackSize(rt, stack_size);
+    if (memory_limit > 0)
+        JS_SetMemoryLimit(rt, (size_t)memory_limit);
+    if (stack_size > 0)
+        JS_SetMaxStackSize(rt, (size_t)stack_size);
```

Comment:
If the rationale is to allow for both `0` and `-1` to mean unlimited, you should make these variables `int64_t` pr `ssize_t`.

Reply:
What would -2 do for example? It feels wrong to allow such values, no?

Reply:
If you make the variables `int64_t` and initialize them to `-1`, you could use `>= 0` to test whether they we specified on the command line.  If the user specifies `--memory-limit=0`, they probably mean to call `JS_SetMemoryLimit(rt, 0);` just in case some other default was in effect.